### ISSUE TITLE
Bug: gzipped reads become corrupted during splitting

### ISSUE THREAD

**Issue by vmikk** [biocore/sortmerna#294]
Hello!

Using gzip-compressed input data causes an error in SortMeRNA.
When SortMeRNA splits the files into chunks (which are stored in `readb`) it loses newline characters and chunks look corrupted.

Reproducible example:
```
cat > p1.fa <<'EOT'
>s1_1
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
>s2_1
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
>s3_1
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
EOT

cat > p2.fa <<'EOT'
>s1_2
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
>s2_2
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
>s3_2
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
EOT

gzip p1.fa p2.fa

sortmerna \
  --ref "smr_v4.3_default_db.fasta" \
  --reads "p1.fa.gz" \
  --reads "p2.fa.gz" \
  --workdir "tst" \
  --fastx --other --paired_in --out2 \
  --threads 1


# Check one of the chunks
zless tst/readb/fwd_0.fa.gz
```

Sequences are erroneously concatenated:
```
>s1_1
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG>s2_1
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG>s3_1
ACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTGACTG
```

If input files are large enough, SortMeRNA will issue an error:
```
[validate:286] ERROR: Read ID: 0_0 Header: >s1_1 Sequence length: 117826171 > 30000 nt
  Please check your reads or contact the authors.
```

With unpacked files there is no error.

SortMeRNA installed via conda:
```
SortMeRNA version 4.3.3
Build Date: May 25 2021
sortmerna_build_git_sha:@09a86c4b9c2f97aceb74af9ef88b38813ea2f956@
sortmerna_build_git_date:@2021/05/25 18:58:26@
```

With kind regards,
Vladimir

**Comment by biocodz** [comment]
Hi. 
Haven't seen this so far. Will try to reproduce. 
Thanks for reporting.

**Comment by biocodz** [comment]
indeed, the split is buggy for gzipped fasta. Working on the fix. Shouldn't take long.

**Comment by biocodz** [comment]
@vmikk 
Hi, please see [fix for issue 294 - pre-release](https://github.com/biocore/sortmerna/releases/tag/v4.3.4-pre)
I created a few new tests that worked OK. Would be great if you could test too prior creating the production release.

**Comment by vmikk** [comment]
Hello @biocodz!
I've tested the pre-release with my data, and everything works flawlessly now.
Thank you very much for fixing the bug!

With kind regards,
Vladimir
```
SortMeRNA version 4.3.4
Build Date: Jul 18 2021
sortmerna_build_git_sha:@90cdf6ce459e9477c5d0de9a5daab025f49f3cc6@
sortmerna_build_git_date:@2021/07/18 12:06:41@
```

**Comment by biocodz** [comment]
Great! 
Thanks for the input. 
Publishing the new release then ...

**Comment by biocodz** [comment]
fixed in [release 4.3.4](https://github.com/biocore/sortmerna/releases/tag/v4.3.4)
